File: Examples/Ruby/extracting-managing-ocr/cloud-storage/extract-ocr-or-hocr-text-from-specific-block-image.rb
require 'aspose_ocr_cloud'

include AsposeStorageCloud
include AsposeOCRCloud
 
configFile = File.expand_path("../../../data/config.json", __FILE__);
configProps = JSON.parse( IO.read(configFile, encoding: 'utf-8') )
data_path = File.expand_path("../../../data/", __FILE__);
out_path = configProps['out_folder'];
  
#ExStart:1
AsposeApp.app_key_and_sid(configProps['api_key'], configProps['app_sid'])
AsposeOCRCloud::Configuration.debugging = false
AsposeOCRCloud::Configuration.temp_folder_path = out_path
AsposeStorageCloud::Configuration.temp_folder_path = out_path

#Instantiate Aspose.Storage API SDK 
@storageApi = StorageApi.new

#Instantiate Aspose.OCR API SDK
@ocrApi = OcrApi.new

#Set the image file name
name = 'Sampleocr.bmp';

#Set the language of the document
language = 'English';

#Set X and Y coordinate to recognize text inside
rect_x = 150;
rect_y = 100;

#Set Width and Height to recognize text inside
rect_width = 1000;
rect_height = 300;

#Set the spelling correction is used
use_default_dictionaries = 'true';


opts = {language: language, rect_x: rect_x, rect_y: rect_y, rect_width: rect_width, rect_height: rect_height, use_default_dictionaries: use_default_dictionaries}

#Upload file to aspose cloud storage 
response = @storageApi.put_create(name, File.open(data_path + "/" << name,"rb"){ |io| io.read } )

#invoke Aspose.OCR Cloud SDK API to extract text and partsinfo from an image
response = @ocrApi.get_recognize_document(name, opts)

if response.status == "OK"
  print "Text :: " + response.text
  print "\n\nExtract OCR or HOCR Text from a specific Block, Done!"
end
#ExEnd:1

File: Examples/Python/extracting-managing-ocr/cloud-storage/extract-ocr-or-hocr-text-from-specific-block-image.py
import asposeocrcloud
from asposeocrcloud.OcrApi import OcrApi
from asposeocrcloud.OcrApi import ApiException

import asposestoragecloud 
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

import ConfigParser

config = ConfigParser.ConfigParser()
config.readfp(open(r'../../data/config.properties'))
apiKey = config.get('AppConfig', 'api_key')
appSid = config.get('AppConfig', 'app_sid')
out_folder = config.get('AppConfig', 'out_folder')
data_folder = "../../data/" #resouece data folder

#ExStart:1
#Instantiate Aspose.Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, False)
storageApi = StorageApi(storage_apiClient)

#Instantiate Aspose.OCR API SDK
api_client = asposeocrcloud.ApiClient.ApiClient(apiKey, appSid, False)
ocrApi = OcrApi(api_client);

#Set the image file name
name = 'Sampleocr.bmp';

#Set the language of the document
language = 'English';

#Set X and Y coordinate to recognize text inside
rectX = 150;
rectY = 100;

#Set Width and Height to recognize text inside
rectWidth = 1000;
rectHeight = 300;

#Set the spelling correction is used
useDefaultDictionaries = 'true';

try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)
    
    #invoke Aspose.OCR Cloud SDK API to extract text and partsinfo from an image
    response = ocrApi.GetRecognizeDocument(name,language=language,rectX=rectX,rectY=rectY,rectWidth=rectWidth,rectHeight=rectHeight,useDefaultDictionaries=useDefaultDictionaries )
    
    if response.Status == 'OK':
        print "Text :: " + response.Text     
        
        print "\n\nExtract OCR or HOCR Text from a specific Block, Done!"
                        
except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message    
#ExEnd:1    